CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -0,0 +1,185 @@
+            }
+
+            if (mentorRepositoryMock != null)
+            {
+                _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorRepositoryMock.Object);
+            }
+
+            if (studentGroupRepositoryMock != null)
+            {
+                _unitOfWorkMock.Setup(x => x.StudentGroupRepository).Returns(studentGroupRepositoryMock.Object);
+            }
+        }
+
+        [Fact]
+        public async Task CreateScheduleAsync_ValidScheduleRequest_ShouldReturnEventOccurrenceDTO()
+        {
+            //Arrange
+            var createScheduleDto = new CreateScheduleDto
+            {
+                Pattern = new PatternForCreateScheduleDTO
+                {
+                    Type = PatternType.Daily,
+                    Interval = 1,
+                    DaysOfWeek = new List<DayOfWeek> { DayOfWeek.Monday },
+                },
+
+                Range = new OccurenceRange
+                {
+                    StartDate = DateTime.Parse("Jan 1, 2021"),
+                    FinishDate = DateTime.Parse("Feb 1, 2021")
+                },
+
+                Context = new ContextForCreateScheduleDTO
+                {
+                    GroupID = 3,
+                    MentorID = 1,
+                    ThemeID = 5
+                }
+            };
+
+            EventOccurrence expected = new EventOccurrence
+            {
+                Pattern = createScheduleDto.Pattern.Type,
+                StudentGroupId = createScheduleDto.Context.GroupID,
+                EventStart = createScheduleDto.Range.StartDate,
+                EventFinish = createScheduleDto.Range.FinishDate.Value,
+                Storage = EventOccuranceStorageParser.GetPatternStorageValue(createScheduleDto.Pattern)
+            };
+
+            var themeRepositoryMock = new Mock<IThemeRepository>();
+
+            themeRepositoryMock.Setup(x => x.IsEntityExistAsync(5)).ReturnsAsync(true);
+
+            var mentorRepositoryMock = new Mock<IMentorRepository>();
+            mentorRepositoryMock.Setup(x => x.IsEntityExistAsync(1)).ReturnsAsync(true);
+
+            var studentGroupRepositoryMock = new Mock<IStudentGroupRepository>();
+            studentGroupRepositoryMock.Setup(x => x.IsEntityExistAsync(3)).ReturnsAsync(true);
+
+            Initialize(createScheduleDto, themeRepositoryMock, mentorRepositoryMock, studentGroupRepositoryMock);
+
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.CreateScheduleAsync(createScheduleDto);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Data.Should().BeEquivalentTo(_mapper.Map<EventOccurrenceDTO>(expected));
+        }
+
+        [Fact]
+        public async Task CreateScheduleAsync_NotValidScheduleRequest_ShouldReturnValidationError()
```

Comment:
It would be good to know what exactly is not valid in the state.
For example, StartDateBiggerThenEndDate, MentorDoesNotExist and so on.

Reply:
Done - https://github.com/ita-social-projects/WhatBackend/pull/406/commits/e66579cd27ba1798016008f148f4cc7ca1322601